pygmt/src/grdlandmask.py

```diff
@@ -0,0 +1,95 @@
+"""
+grdlandmask - Create a "wet-dry" mask grid from shoreline data base
+"""
+
+import xarray as xr
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import (
+    GMTTempFile,
+    build_arg_string,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+
+
+@fmt_docstring
+@use_alias(
+    G="outgrid",
+    R="region",
+    I="increment",
+    r="registration",
+)
+@kwargs_to_strings(R="sequence")
+def grdlandmask(**kwargs):
+    r"""
+    Create a grid file with set values for land and water.
+
+    Read the selected shoreline database and create a grid to specify which
+    nodes in the specified grid are over land or over water. The nodes defined
+    by the selected region and lattice spacing
+    will be set according to one of two criteria: (1) land vs water, or
+    (2) the more detailed (hierarchical) ocean vs land vs lake
+    vs island vs pond.
+
+    Full option list at :gmt-docs:`grdlandmask.html`
+
+    {aliases}
+
+    Parameters
+    ----------
+    outgrid : str or None
+        The name of the output netCDF file with extension .nc to store the grid
+        in.
+    increment : str
+        *xinc*\ [**+e**\|\ **n**][/\ *yinc*\ [**+e**\|\ **n**]].
+        *x_inc* [and optionally *y_inc*] is the grid spacing. **Geographical
+        (degrees) coordinates**: Optionally, append a increment unit. Choose
+        among **m** to indicate arc minutes or **s** to indicate arc seconds.
+        If one of the units **e**, **f**, **k**, **M**, **n** or **u** is
+        appended instead, the increment is assumed to be given in meter, foot,
+        km, mile, nautical mile or US survey foot, respectively, and will be
+        converted to the equivalent degrees longitude at the middle latitude
+        of the region. If *y_inc* is given but set to 0 it will be reset equal
+        to *x_inc*; otherwise it will be converted to degrees latitude. **All
+        coordinates**: If **+e** is appended then the corresponding max
+        *x* (*east*) or *y* (*north*) may be slightly adjusted to fit exactly
+        the given increment [by default the increment may be adjusted slightly
+        to fit the given domain]. Finally, instead of giving an increment you
+        may specify the *number of nodes* desired by appending **+n** to the
+        supplied integer argument; the increment is then recalculated from the
+        number of nodes, the *registration*, and the domain. The resulting
+        increment value depends on whether you have selected a
+        gridline-registered or pixel-registered grid.
```

Comment:
This would be a good option to include as standardized text, similar to https://github.com/GenericMappingTools/pygmt/pull/1186. Here is a list of all the GMT modules that use identical text to describe this parameter, so PyGMT could similarly use standardized text in its docs.
```
blockmean
blockmedian
blockmode
gmtbinstats
grdblend
grdfilter
grdlandmask
grdmask
grdmath
grdsample
mask_common_
nearneighbor
sph2grd
sphdistance
sphinterpolate
supplements/geodesy/earthtide
supplements/geodesy/gpsgridder
supplements/potential/grdgravmag3d
supplements/potential/grdseamount
supplements/potential/talwani3d
supplements/spotter/grdspotter
supplements/spotter/hotspotter
supplements/spotter/polespotter
surface
triangulate
xyz2grd
```

Reply:
I wouldn't put this huge block of text for spacing (-I) in `fmt_docstring` like #1186 to be honest, but maybe we could add it to the COMMON_OPTIONS dictionary in [decorators.py](https://github.com/GenericMappingTools/pygmt/blob/master/pygmt/helpers/decorators.py) (even if it isn't exactly a GMT common option)?

Reply:
These seems like a good option to me. There are already multiple non-common options in COMMON_OPTIONS anyways.

Reply:
Ok, I'll work on a PR for this after lunch.

Reply:
Sorry for slowing you down. If only we had more online hours in common.